lisa/runners/lisa_runner.py

```diff
@@ -392,6 +401,7 @@ def _delete_environment_task(
             in [
```

Comment:
I tried this, but when set failed to keep_environment and case failed, the env still be deleted
```
        if environment.status == EnvironmentStatus.Prepared and not environment.is_in_use:
            # The prepared only environment doesn't need to be deleted. It may cause platform fail to delete non-existing environment.
            environment.status = EnvironmentStatus.Deleted
        else:
            try:
                self.platform.delete_environment(environment)
            except Exception as identifier:
                self._log.debug(
                    f"error on deleting environment '{environment.name}': {identifier}"
                )
```

Reply:
ok, it looks if the state is deleted already, it should go with first logic too. Let me know, if there is other cases.

Reply:
I add EnvironmentStatus.Deleted situation into if, so far, test results are expected.